### PR TITLE
Updated net7 isolated AI example to use Worker.ApplicationInsights 1.0.0

### PR DESCRIPTION
**Pull request by p-m-j**

### CODE REVIEW

File: Func.Isolated.Net7.With.AI/Func.Isolated.Net7.With.AI.csproj [hunk 1]
```diff
@@ -15,8 +15,9 @@
     </Content>
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.21.0" />
```
Comment (LockTar) [inline Func.Isolated.Net7.With.AI/Func.Isolated.Net7.With.AI.csproj:18]: Do you have any documentation reference about this package reference?

File: Func.Isolated.Net7.With.AI/Program.cs [hunk 2]
```diff
@@ -6,16 +6,12 @@
...
-        // Documented here because it is still preview: https://github.com/Azure/azure-functions-dotnet-worker/pull/944#issue-1282987627
-        builder
-            .AddApplicationInsights()
-            .AddApplicationInsightsLogger();        
-    })
+    .ConfigureFunctionsWorkerDefaults()
     .ConfigureServices((ctx, serviceProvider) =>
     {
+        serviceProvider.AddApplicationInsightsTelemetryWorkerService();
+        serviceProvider.ConfigureFunctionsApplicationInsights();
```
Comment (LockTar) [inline Func.Isolated.Net7.With.AI/Program.cs:18]: Do you have any documentation reference of these serviceProvider extensions?

**Review by LockTar — CHANGES_REQUESTED** [review]
Thank you for creating this PR. I saw the release but didn't had time yet to check it.
Because this is a sample project with extra documentation I would like to add comments of every line about what the idea behind it is. 

Hopefully you understand.

I think you have those references because otherwise you where not able to create this PR :-).

File: Func.Isolated.Net7.With.AI/Program.cs (hunk 2 above)
Comment (p-m-j) [reply Func.Isolated.Net7.With.AI/Program.cs:18]: MS docs cover this, https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#application-insights

File: Func.Isolated.Net7.With.AI/Func.Isolated.Net7.With.AI.csproj (hunk 1 above)
Comment (p-m-j) [reply Func.Isolated.Net7.With.AI/Func.Isolated.Net7.With.AI.csproj:18]: This one is covered here https://learn.microsoft.com/en-us/azure/azure-monitor/app/worker-service

File: Func.Isolated.Net7.With.AI/Program.cs (hunk 2 above)
Comment (LockTar) [reply Func.Isolated.Net7.With.AI/Program.cs:18]: Thanks. I see that https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#application-insights is mentioning this package reference as well

File: Func.Isolated.Net7.With.AI/Func.Isolated.Net7.With.AI.csproj (hunk 1 above)
Comment (LockTar) [reply Func.Isolated.Net7.With.AI/Func.Isolated.Net7.With.AI.csproj:18]: Thanks. I see that https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#application-insights is mentioning this package reference as well

**Review by LockTar — APPROVED** [review]

**Comment by LockTar** [comment]
@p-m-j Have you tested you PR? Because I'm testing right now and got al kind of duplicate messages... I think I'm going to revert your PR. Or at least part of it.

**Comment by p-m-j** [comment]
~@LockTar yes I get the same behaviour, an unexpected trace entry from the worker codebase with category 'Host.Function.Console' and another with the expected fully qualified name.~ 

~This seems to be at odds with the documentation [here](https://learn.microsoft.com/en-us/azure/azure-functions/configure-monitoring?tabs=v2)~

~> When custom application logs are sent directly, the host no longer be emits them, and host.json no longer controls their behavior.~

~I had made a comment questioning the behaviour in another issue in the [azure-functions-dotnet-worker](https://github.com/Azure/azure-functions-dotnet-worker) repo but wasn't happy with how I worded it and nuked the comment.~

The issue I had here was not related to this repo.

**Pull request merged**